lib/battleship.rb

```diff
@@ -28,19 +40,59 @@ def start
       # puts result of computer's shot
       # check if all user's ships are sunk
       # if yes, change @game_over to "computer"
-    end
-
-    end_game
-  end
+  #   end
+  #
+  #   end_game
+  # end
 
   def setup
     # set up two boards
+    players = [@user, @computer]
     # @boards << computer board
     # @boards << user board
     # ask @computer to place ships
     # ask @user to place ships
+
+    players.each do |player|
+      player.ships.each do |ship|
+        coordinates = []
+        puts "Place #{ship.name} (#{ship.length} cells)"
+        puts "Enter first coordinate"
+        first_coordinate = gets.chomp.capitalize
+        coordinates << first_coordinate
+        (ship.length - 1).times do
+          puts 'Next Coordinate?'
+          next_coordinate = gets.chomp.capitalize
+          coordinates  << next_coordinate
+        end
+        validation_checks(player.own_board, ship, coordinates)
+      end
+    end
   end
 
+  def validation_checks(board, ship, coordinates)
```

Comment:
Let's do this inside the Player class - they can do validation as part of their turn